packages/gestalt/src/TypeaheadOption.js

```diff
@@ -0,0 +1,92 @@
+// @flow strict
+import * as React from 'react';
+import { useState } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import Box from './Box.js';
+import Text from './Text.js';
+import styles from './Touchable.css';
+import focusStyle from './TypeaheadInputField.css';
+import getRoundingClassName from './getRoundingClassName.js';
+
+type OptionObject = {|
+  label: string,
+  value: string,
+|};
+
+type OptionProps = {|
+  index: number,
+  option: OptionObject,
+  selected?: OptionObject | null,
+  searchField: string,
+  handleSelect: OptionObject => void,
+|};
+
+export default function Option({
+  index,
+  option,
+  selected,
+  searchField,
+  handleSelect,
+}: OptionProps): React.Node {
+  // Determine if the option is the current selected item
+  const isSelectedItem = JSON.stringify(option) === JSON.stringify(selected);
+
+  // Highlight the current selected item
+  const [hover, setHover] = useState(isSelectedItem);
+
+  const handleOnTap = () => {
+    if (handleSelect) handleSelect(option);
+  };
+
+  const className = classnames(styles.touchable, getRoundingClassName(2), {
+    [styles.fullWidth]: true,
+    [styles.pointer]: true,
+  });
+
+  return (
+    <div
+      className={`${className} typeahead-option ${
```

Comment:
@simkessy could we
* remove `typeahead-option` - we do not allow static classnames
* make `isSelectedItem ? focusStyle.typeaheadOptionFocus : ''` part of the classnames function above